### PR TITLE
[Bug 20858] Use an asset catalog for iOS icons

### PR DESCRIPTION
**Pull request by livecodepanos**

### CODE REVIEW

**Comment by livecodepanos** [comment]
Depends on https://github.com/livecode/livecode-ide/pull/1890

File: ide-support/revsaveasiosstandalone.livecodescript [hunk 1]
```diff
@@ -747,21 +747,98 @@ end revSaveAsMobileDeployer
...
+         "iPad Pro 12.9 icon,ipad,83.5x83.5,2x,Icon-App-83.5x83.5@2x.png" \
+         into sIconsList
+   
+   local tAssetsFolder, tAppIconFolder
+   put tempname() into tAssetsFolder
+   create folder  tAssetsFolder
+   put there is a folder tAssetsFolder
+   put tAssetsFolder & slash & "AppIcon.appiconset" into tAppIconFolder
+   create folder tAppIconFolder
+   put there is a folder tAppIconFolder
```
Comment (livecodeali) [inline ide-support/revsaveasiosstandalone.livecodescript]: These `put there is a folder tAssetsFolder` and `put there is a folder tAppIconFolder` should probably be removed

File: ide-support/revsaveasiosstandalone.livecodescript [hunk 2]
```diff
@@ -747,21 +747,98 @@ end revSaveAsMobileDeployer
...
+   {
+      "idiom" : "ios-marketing",
+      "size" : "1024x1024",
+      "scale" : "1x",
+      "filename" : "iTunesArtwork@2x.png"
+    }
+   */
+   local sIconsList 
+   put  \
+         "appstore icon,ios-marketing,1024x1024,1x,ItunesArtwork@2x.png" & return & \
```
Comment (livecodeali) [inline ide-support/revsaveasiosstandalone.livecodescript]: Is this meant to be scale `1x` but an `@2x.png` image file?

File: ide-support/revsaveasiosstandalone.livecodescript [hunk 3]
```diff
@@ -747,21 +747,98 @@ end revSaveAsMobileDeployer
...
+   if the result is not empty then
+      throw "could not create json file '" & it & "'"
+   end if
+   
+   local tCompiledAssetsFolder
+   put tempname() into tCompiledAssetsFolder
+   create folder  tCompiledAssetsFolder
+   
+   -- create the Assets.car file
+   get shell("xcrun actool " &  tAssetsFolder & " --compile " & tCompiledAssetsFolder & " --platform iphoneos --minimum-deployment-target 9.0 --app-icon AppIcon --output-partial-info-plist " & tCompiledAssetsFolder &"/partial.plist")
```
Comment (livecodeali) [inline ide-support/revsaveasiosstandalone.livecodescript]: could you put a couple of line continuations here to make this a bit more readable?

File: ide-support/revsaveasiosstandalone.livecodescript (hunk 2 above)
Comment (livecodepanos) [reply ide-support/revsaveasiosstandalone.livecodescript]: Yes, although it seems strange, this is intended. This is the exact entry that is created if you use Xcode's interface to create the asset catalog.

File: ide-support/revsaveasiosstandalone.livecodescript (hunk 1 above)
Comment (livecodepanos) [reply ide-support/revsaveasiosstandalone.livecodescript]: Oops I forgot to remove them :)

File: ide-support/revsaveasiosstandalone.livecodescript (hunk 3 above)
Comment (livecodepanos) [reply ide-support/revsaveasiosstandalone.livecodescript]: sure :)

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 8804f42

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 8804f4293a43c5696e57d0d145875145e586aded

**Comment by livecode-vulcan** [comment]
😎 test success 8804f4293a43c5696e57d0d145875145e586aded

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1169)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1185)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1159)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/112)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/112)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1164)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1186)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1169)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1185)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1168)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1178)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1202)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1163)

**Pull request merged**